debug/process/debugger.go

```diff
@@ -0,0 +1,162 @@
+package process
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"sync"
+	"time"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/config"
+)
+
+const minAcceptedValue = 1
+
+const buffSize = 100 * 1024 * 1024 // 100MB
+var log = logger.GetOrCreate("debug/process")
+
+type processDebugger struct {
+	timer                   *time.Timer
+	mut                     sync.RWMutex
+	lastCheckedBlockRound   int64
+	lastCommittedBlockRound int64
+	cancel                  func()
+	goRoutinesDumpHandler   func()
+	logChangeHandler        func()
+	pollingTime             time.Duration
+	logLevel                string
+	dumpGoRoutines          bool
+}
+
+// NewProcessDebugger creates a new debugger instance used to monitor the block process flow
+func NewProcessDebugger(config config.ProcessDebugConfig) (*processDebugger, error) {
+	err := checkConfigs(config)
+	if err != nil {
+		return nil, err
+	}
+
+	pollingTime := time.Duration(config.PollingTimeInSeconds) * time.Second
+	d := &processDebugger{
+		timer: time.NewTimer(pollingTime),
+
+		pollingTime:    pollingTime,
+		logLevel:       config.LogLevelChanger,
+		dumpGoRoutines: config.GoRoutinesDump,
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	d.cancel = cancel
+	d.goRoutinesDumpHandler = func() {
```

Comment:
why does this function have to be declared? 
it can be expressed as a normal function.

Reply:
it was done like this in order to thoroughly test the component (in tests, I swapped these function pointers with testing functions)

Reply:
created method